### DISCUSSION TITLE
Time Evolution of Lindblad Master equation/ non-hermitian hamiltonian

### DISCUSSION THREAD

**Discussion by manthan-badbaria** [jcmgray/quimb#193] [Q&A]
Hi,
I am trying to evolve the Lindblad master equation in vectorized form using TEBD:
\rho^{.} = L\rho, where \rho , L are superket and super operator respectively 
and L= -i(H\kron I + I\kron H) + \sum_m 0.5*(2L_m\kron L_m.conj() - L^{dagger}_m@L_m \kron I - I\kron L^{T}_m@L_m.conj() )
where L_m is Lindblad operator at each site.
Now I want to evolve a vectorised density matrix(fusing the physical indices of an mpo to convert it into an mps) using the above lindbladian. I use the function LocalHam1d with 2 site interaction terms and L_m as single site terms with dimension 4 as we are evolving in double space.
I benchmark my results with ED calculation for smaller systems and they do not match. 
I think I have figured the issue which is that LocalHam1d treats the hamiltonian as hermitian and uses eigh function to calculate singular values, where as it should use expm to account for non-hermitian hamiltonian. Similar tensor libraries like tenpy have incorporated this in their TEBD algorithm. Is there a way to do the same using TEBD algorithm in quimb?

**Comment by jcmgray** [comment]
Hi @manthan-badbaria, yes sounds like it might be as simple as changing `_expm_cached` function in `tensor_arbgeom_tebd`. You could try and see if that fixes things and if so we can add to `quimb`.

**Reply by manthan-badbaria** [reply]
Hi @jcmgray, I tried modifying the _expm_cached function using expm() and it works very well. It agrees with my ed results for non-hermitian hamiltonian evolution.
